test/unit/PerceptronPredictorTest.cc

```diff
@@ -0,0 +1,148 @@
+#include "MockInstruction.hh"
+#include "gtest/gtest.h"
+#include "simeng/PerceptronPredictor.hh"
+
+namespace simeng {
+
+class PerceptronPredictorTest : public testing::Test {
+ public:
+  PerceptronPredictorTest() : uop(new MockInstruction), uopPtr(uop) {
+    uop->setInstructionAddress(0);
+  }
+
+ protected:
+  MockInstruction* uop;
+  std::shared_ptr<Instruction> uopPtr;
+};
```

Comment:
Please add a unit test for the flush() function. An example for the GenericPredictor can be found in the unit test PR